game-core/src/test/java/org/triplea/util/PointFileReaderWriterTest.java

```diff
@@ -505,13 +498,9 @@ void shouldWriteOnePointPerName() throws Exception {
       final String content =
           writeToString(os -> PointFileReaderWriter.writeOneToOne(os, pointsByName));
 
-      assertThat(
-          content,
-          is(
-              ""
-                  + "United Kingdom  (1011,1021) \r\n"
-                  + "Germany  (2011,2021) \r\n"
-                  + "Eastern United States  (3011,3021) "));
+      assertThat(content, containsString("United Kingdom  (1011,1021)"));
+      assertThat(content, containsString("Germany  (2011,2021)"));
+      assertThat(content, containsString("Eastern United States  (3011,3021)"));
```

Comment:
Line endings are somewhat important to check.
If everything is in the same line that's a syntax error

Reply:
The ordering of the three lines is non-deterministic. It was actually a major flaw in the test previously that a non-deterministic ordering was being relied upon. We could assert that we see 2 counts of the line terminator, though it may not mean much as it could appear anywhere in the string.

Reply:
Hmm we could change the code to generate files with unix-style line endings, then we could check the line endings for every line

Reply:
> It was actually a major flaw in the test previously that a non-deterministic ordering was being relied upon.

I believe this test did not previously rely on a non-deterministic ordering.  Guava's [`ImmutableMap#of()`](https://guava.dev/releases/26.0-jre/api/docs/com/google/common/collect/ImmutableMap.html#of-K-V-K-V-K-V-K-V-K-V-) factory methods guarantee that the returned map entries are iterated in the same order as the arguments.  Whereas, the Java `Map#of()` factory methods do not make the same guarantee.

Reply:
I did not know that about `ImmutableMap#of`. I'm still concerned:
- the API takes a `Map`, typically the ordering is not expected to be guaranteed
- the test code assigned to a `Map`, we're relying on the contract of a specific implementation which as was the case here, is error prone as the implementation was changed and caused a breakage
- the test assumes a specific implementation, notably that the elements would be processed in-order, that does not seem to be required for correctness
- relying on specific nuances of Guava seems to increase barrier to entry

Are there any ideas of how to avoid the guava immutable map here? I don't think the test coverage for the line endings is actually that profound, we're testing the `collect` method on streams:

```
        mapping.entrySet().stream()
            .map(entry -> entry.getKey() + " " + pointToString(entry.getValue()))
            .collect(Collectors.joining("\r\n")),
```

Reply:
> Are there any ideas of how to avoid the guava immutable map here?

I guess you could change the implementation to be deterministic instead of the test.  That is, sort the stream by key and simply document that fact in the Javadocs.  I'm not aware of any downstream consumer that would be affected by that change.

IIRC, the reason the above proposal wasn't implemented at the time the original tests were written is because the entire set of `PointFileReaderWriter` tests were added as a single group of characterization tests prior to doing some refactoring (see #2889 and #2924).  So, by definition, the implementation wasn't supposed to be changed before the tests were complete.  The idea to go back and make the implementation deterministic instead of the test was probably just lost in the follow-up noise.

Reply:
Certainly a number of issues present here, for the sake of moving forward, went back to `ImmutableMap#of` with: df0e2ec58